MainWindow.cpp

```diff
@@ -564,10 +564,19 @@ void MainWindow::CreateResource(TypeCase typeCase) {
   refl->MutableMessage(&child, field);
 
   // find a unique name for the new resource
-  child.set_name(resourceMap->CreateResourceName(&child).toStdString());
+  QString resourceName = resourceMap->CreateResourceName(&child);
+  child.set_name(resourceName.toStdString());
+
   // release ownership of the new child to its parent and the tree
   auto index = this->treeModel->addNode(child, _ui->treeView->currentIndex());
   treeModel->triggerNodeEdit(index, _ui->treeView);
+
+  // add new resource with created name, helps in creating another unique name
+  MessageModel* messageModel = treeModel->IndexToNode(index)->BackingModel()->TryCastAsMessageModel();

```

Comment:
This chain actually has room for a bunch of nullptrs in it. I believe both IndexToNode and BackingModel can return null, and having `this` as nullptr is UB (it's legal to optimize out checks for `this == null`).